internal/api/api.go

```diff
@@ -114,29 +114,38 @@ func handleHealthcheck(c *gin.Context) {
 // @Failure      500           {object}  string  "Server Error"
 // @Router       /api/submit/tx [post]
 func handleSubmitTx(c *gin.Context) {
+	// First, initialize our configuration and loggers
 	cfg := config.GetConfig()
 	logger := logging.GetLogger()
-	// Read transaction from request body
-	rawTx, err := ioutil.ReadAll(c.Request.Body)
+	// Read raw transaction bytes from the request body and store in a byte array
+	txRawBytes, err := ioutil.ReadAll(c.Request.Body)
 	if err != nil {
+		// Log the error, return an error to the user, and increment failed count
 		logger.Errorf("failed to read request body: %s", err)
-		c.String(500, "failed to request body")
+		c.String(400, "failed to read request body")
```

Comment:
A `400` response implies a client problem, but an error here would be due to a failure of the underlying `read()` call, which either means our client is already gone or something "system" failed, which should be a `500`

Reply:
Fair enough. I'll change it back.